pyqt-apps/siriushla/as_di_scrns/main.py

```diff
@@ -425,19 +446,70 @@ def _saveCalibrationGrid_thread(self):
             state = self.SiriusLedState_CamEnbl.state
 
         # Change ROI to original size
-        self.PyDMSpinbox_ImgROIHeight.send_value_signal[float].emit(roi_h)
-        self.PyDMSpinbox_ImgROIWidth.send_value_signal[float].emit(roi_w)
-        self.PyDMSpinbox_ImgROIOffsetX.send_value_signal[float].emit(
-            roi_offsetx)
-        self.PyDMSpinbox_ImgROIOffsetY.send_value_signal[float].emit(
-            roi_offsety)
+        self.ch_ImgROIHeight.send_value_signal[float].emit(roi_h)
+        self.ch_ImgROIWidth.send_value_signal[float].emit(roi_w)
+        self.ch_ImgROIOffsetX.send_value_signal[float].emit(roi_offsetx)
+        self.ch_ImgROIOffsetY.send_value_signal[float].emit(roi_offsety)
 
         # Enable camera acquisition
         self.PyDMStateButton_CamEnbl.send_value_signal[int].emit(1)
 
         # Enable showing saved grid
         time.sleep(0.1)
         self.checkBox_showgrid.setEnabled(True)
+        self.save_files.emit()
+
+    def _saveGridLocalFiles(self):
+        home = os.path.expanduser('~')
+        folder_month = datetime.now().strftime('%Y-%m')
+        folder_day = datetime.now().strftime('%Y-%m-%d')
+        path = os.path.join(
+            home, 'Desktop', 'screens-iocs', folder_month, folder_day)
+        if not os.path.exists(path):
+            os.makedirs(path)
+        fn, _ = QFileDialog.getSaveFileName(
+            self, 'Save Grid As...', path + '/' + self.device +
+            datetime.now().strftime('_%Y-%m-%d_%Hh%Mmin'), '*.txt')
+        if not fn:
+            return False
+        if not fn.endswith('.txt'):
+            fn += '.txt'
+
+        path_default = os.path.join(
+            home, 'Desktop', 'screens-iocs', 'default')
+        if not os.path.exists(path_default):
+            os.makedirs(path_default)
+        fn_default = path_default + '/' + self.device + '.txt'
+
+        grid = self.image_view.calibrationGrid
+        width = self.image_view.imageWidth
+        data = np.append(width, grid)
+        np.savetxt(fn, data)
+        np.savetxt(fn_default, data)
+
+    def _loadCalibrationGrid(self, default=False):
+        home = os.path.expanduser('~')
+        if not default:
+            folder_month = datetime.now().strftime('%Y-%m')
+            path = os.path.join(
+                home, 'Desktop', 'screens-iocs', folder_month)
+            fn, _ = QFileDialog.getOpenFileName(
+                self, 'Load Grid...', path, '*.txt')
+            if not fn:
+                return
+        else:
+            path = os.path.join(
+                home, 'Desktop', 'screens-iocs', 'default')
+            fn = path + '/' + self.device + '.txt'
+
+        try:
+            data = np.loadtxt(fn)
+            self.image_view.calibrationGrid = data
+        except Exception:
```

Comment:
wouldn't it be better to show a message in case of failure? I understand that you don't want to raise this message at object initialization, so maybe you could implement a flag indicating whether to show the message or not...

Reply:
You're right! I've implemented it!